app-ios/Sources/SessionFeature/SessionView.swift

```diff
@@ -1,39 +1,157 @@
+import Assets
 import ComposableArchitecture
 import Model
+import Strings
 import SwiftUI
+import Theme
 
 public struct SessionState: Equatable {
-    public var timetableItem: TimetableItem
+    public var timetableItemWithFavorite: TimetableItemWithFavorite
+    public var showShareSheet: Bool = false
 
-    public init(timetableItem: TimetableItem) {
-        self.timetableItem = timetableItem
+    public init(timetableItemWithFavorite: TimetableItemWithFavorite) {
+        self.timetableItemWithFavorite = timetableItemWithFavorite
     }
 }
 
 public enum SessionAction {
+    case tapCalendar(TimetableItem)
+    case tapFavorite(TimetableItemId, Bool)
+    case tapShare
+    case nothing
 }
 
 public struct SessionEnvironment {
-    public init() {}
+    public let sessionsRepository: SessionsRepository
+
+    public init(sessionsRepository: SessionsRepository) {
+        self.sessionsRepository = sessionsRepository
+    }
 }
 
-public let sessionReducer = Reducer<SessionState, SessionAction, SessionEnvironment> { _, action, _ in
+public let sessionReducer = Reducer<SessionState, SessionAction, SessionEnvironment> { state, action, environment in
 
     switch action {
+    case .tapCalendar(let item):
+        return .run { @MainActor _ in
+            // TODO: Add event using EventKit
+        }
```

Comment:
I'm not 100% sure how to add event using EventKit. So instead I've added a TODO instead for future Issue.